### PR TITLE
Very Robust Consolidate Body Parts Function

### PR DESCRIPTION
**Pull request by bombasticSlacks**
#### Summary
Features "Similar sub parts are grouped in item UI"

#### Purpose of change
Unhardcoding some body part stuff in the UI. Robustifying the simple UI fix in #56476 
Also we had a problem with items where some really simple items that covered a lot of locations looked **really complicated** while really complex items that covered very few locations always looked really simple. Now complexity is properly represented.

#### Describe the solution
- Moves consolidate into body part so it can reference body parts not just sub parts.
- Tries to combine subparts into limbs then limbs into pairs, then sub limbs into pairs.
- Returned list is alphabetized

This is then used to consolidate body parts in the UI for items. With this I had to split up things related to only body parts from things related to sub parts. This in general will always make item entries shorter *and* if it doesn't it is because new info is now displayed to the player that was hidden before and shouldn't have been.

#### Testing
Fixed like all the UI tests so they apply to this. Since they work this works 😆

#### Additional context
Some entries

Stuff like the emergency blanket are much shorter since each entry is collapsed into one data set 
image

Even stuff like the survivor suits are much simpler now since the same info isn't repeated for each of arms, legs and torso
image

Something like the heavy ballistic vest is an example of something that is complicated but now gives more correct info to the player.
image

### CODE REVIEW

File: src/translation.cpp [hunk 1]
```diff
@@ -416,3 +416,8 @@ std::string operator+( const translation &lhs, const translation &rhs )
 {
     return lhs.translated() + rhs.translated();
 }
+
+bool operator<( const translation &lhs, const translation &rhs )
```
Comment (Qrox) [inline src/translation.cpp:420]: Please use `translation::translated_lt` for localized comparison. Defining `<` as the localized one will be inconsistent with `==` and `!=` defined in `class translation`.

Comment (bombasticSlacks) [reply src/translation.cpp:420]: Thanks for taking the time to review. That makes sense. 

I'm pretty sure localized_comparator will work, if scoping ends up being an issue I'll figure something out.

I'm on vacation for the next week and can't compile so I'll probably not get to this but just wanted to let you know I'm not ignoring the suggestion 😄

**Review by dseguin — APPROVED** [review]
Merged & tested locally, looks good! If the changes in the unit test are anything to go by, I don't see any issues.

File: src/translation.h [hunk 2]
```diff
@@ -221,5 +221,6 @@ std::ostream &operator<<( std::ostream &out, const translation &t );
 std::string operator+( const translation &lhs, const std::string &rhs );
 std::string operator+( const std::string &lhs, const translation &rhs );
 std::string operator+( const translation &lhs, const translation &rhs );
+bool operator<( const translation &lhs, const translation &rhs );
```
Comment (Qrox) [inline src/translation.h:224]: Not sure why the compilers were not complaining about this but it should be removed,

**Pull request merged**